edgeless_inabox/src/bin/edgeless_inabox.rs

```diff
@@ -1,44 +1,221 @@
+use std::{collections::HashMap, path::Path};
+
+use anyhow::anyhow;
 use clap::Parser;
+use edgeless_con::{EdgelessConOrcConfig, EdgelessConResourceConfig};
+use edgeless_dataplane::core::EdgelessDataplanePeerSettings;
+use edgeless_inabox::InABoxConfig;
+use edgeless_node::EdgelessNodeSettings;
+use edgeless_orc::EdgelessOrcNodeConfig;
+use std::fs;
+use uuid::Uuid;
 
 #[derive(Debug, clap::Parser)]
 #[command(long_about = None)]
 struct Args {
-    #[arg(short, long, default_value_t = String::from("node.toml"))]
-    node_config_file: String,
-    #[arg(short, long, default_value_t = String::from("orchestrator.toml"))]
-    orc_config_file: String,
-    #[arg(short, long, default_value_t = String::from("balancer.toml"))]
-    bal_config_file: String,
-    #[arg(short, long, default_value_t = String::from("controller.toml"))]
-    con_config_file: String,
-    #[arg(short, long)]
+    /// Legacy option to generate configs for minimal cluster with only one
+    /// worker node. Configs are generated at the root of the repository.
+    #[arg(long, short)]
     templates: bool,
+    /// Runs the edgeless-in-a-box with the specified number of worker nodes and
+    /// saves the config files to configs/ directory for reference. Currently,
+    /// it only supports generating new configs on each use, not reusing old files.
+    #[arg(long, default_value_t = 1)]
+    num_of_nodes: i32,
 }
 
 fn main() -> anyhow::Result<()> {
     env_logger::init();
-
     let args = Args::parse();
-    if args.templates {
+    let config: InABoxConfig;
+
+    if args.templates || args.num_of_nodes == 1 {
+        log::info!("Generating default templates for one node");
         edgeless_api::util::create_template("node.toml", edgeless_node::edgeless_node_default_conf().as_str())?;
         edgeless_api::util::create_template("orchestrator.toml", edgeless_orc::edgeless_orc_default_conf().as_str())?;
         edgeless_api::util::create_template("balancer.toml", edgeless_bal::edgeless_bal_default_conf().as_str())?;
         edgeless_api::util::create_template("controller.toml", edgeless_con::edgeless_con_default_conf().as_str())?;
+        config = InABoxConfig {
+            node_conf_files: vec!["node.toml".to_string()],
+            orc_conf_file: "orchestrator.toml".to_string(),
+            bal_conf_file: "balancer.toml".to_string(),
+            con_conf_file: "controller.toml".to_string(),
+        };
         return Ok(());
+    } else if args.num_of_nodes > 0 {
```

Comment:
How are you planning to use this?
Additional nodes on the same device only make sensor for e.g. controller / orchestrator work, not for the typical inabox usecase of just having a quick development environment.

Reply:
This was added to address exactly the need to have a quick development environment inabox, but with multiple nodes.